src/com/google/javascript/jscomp/gwt/client/GwtRunner.java

```diff
@@ -242,32 +254,57 @@ private static native JavaScriptObject createError(String file, String descripti
   /**
    * Generates the output code, taking into account the passed {@code outputWrapper}.
    */
-  private static String writeOutput(Compiler compiler, String outputWrapper) {
+  private static ModuleOutput writeOutput(Compiler compiler, Flags flags) {
+    ArrayList<File> outputFiles = new ArrayList<>();
+    ModuleOutput output = new ModuleOutput();
+
+    File file = new File();
+    file.path = flags.jsOutputFile;
+
     String code = compiler.toSource();
-    if (outputWrapper == null) {
-      return code;
-    }
+    String prefix = "";
+    String postfix = "";
+    if (flags.outputWrapper != null) {
+      String marker = null;
+      int pos = flags.outputWrapper.indexOf(OUTPUT_MARKER_JS_STRING);
+      if (pos != -1) {
+        // With jsstring, run SourceCodeEscapers (as per AbstractCommandLineRunner).
+        code = SourceCodeEscapers.javascriptEscaper().escape(code);
+        marker = OUTPUT_MARKER_JS_STRING;
+      } else {
+        pos = flags.outputWrapper.indexOf(OUTPUT_MARKER);
+        if (pos != -1) {
+          marker = OUTPUT_MARKER;
+        }
+      }
 
-    String marker;
-    int pos = outputWrapper.indexOf(OUTPUT_MARKER_JS_STRING);
-    if (pos != -1) {
-      // With jsstring, run SourceCodeEscapers (as per AbstractCommandLineRunner).
-      code = SourceCodeEscapers.javascriptEscaper().escape(code);
-      marker = OUTPUT_MARKER_JS_STRING;
-    } else {
-      pos = outputWrapper.indexOf(OUTPUT_MARKER);
-      if (pos == -1) {
-        return code;  // neither marker could be found, just return code
+      if (marker != null) {
+        prefix = flags.outputWrapper.substring(0, pos);
+        SourceMap sourceMap = compiler.getSourceMap();
+        if (sourceMap != null) {
+          sourceMap.setWrapperPrefix(prefix);
+        }
       }
-      marker = OUTPUT_MARKER;
+      postfix = flags.outputWrapper.substring(pos + marker.length());
     }
-
-    String prefix = outputWrapper.substring(0, pos);
-    SourceMap sourceMap = compiler.getSourceMap();
-    if (sourceMap != null) {
-      sourceMap.setWrapperPrefix(prefix);
+    if (flags.createSourceMap) {
+      if (flags.createSourceMap) {
+        StringBuilder b = new StringBuilder();
+        try {
+          compiler.getSourceMap().appendTo(b, flags.jsOutputFile);
+        } catch (IOException e) {
+          // ignore
+        }
+        file.sourceMap = b.toString();
+      }
     }
-    return prefix + code + outputWrapper.substring(pos + marker.length());
+
+    file.src = prefix + code + postfix;
+    outputFiles.add(file);
+    output.compiledFiles = outputFiles.toArray(new File[outputFiles.size()]);
```

Comment:
Will this only be of size one right now? Should we update the JS wrappers to recognize multiple file output rather than relying on compiledCode?

Reply:
I think the existing wrappers just assume a single file out and get the source via `compiledCode`. I wrote this originally to be somewhat filename agnostic.
Just keep an eye on it when you merge in the gulp/grunt/etc wrappers.